lib/Subscriber.js

```diff
@@ -0,0 +1,489 @@
+const EventEmitter = require('events').EventEmitter;
+const JsSIP_C = require('./Constants');
+const Utils = require('./Utils');
+const Grammar = require('./Grammar');
+const debug = require('debug')('JsSIP:Subscriber');
+const debugerror = require('debug')('JsSIP:ERROR:Subscriber');
+
+debugerror.log = console.warn.bind(console);
+
+/**
+ * Termination code 
+ */
+const C = {
+  SUBSCRIBE_RESPONSE_TIMEOUT      : 0, 
+  SUBSCRIBE_TRANSPORT_ERROR       : 1, 
+  SUBSCRIBE_NON_OK_RESPONSE       : 2, 
+  SUBSCRIBE_FAILED_AUTHENTICATION : 3,
+  UNSUBSCRIBE_TIMEOUT             : 4, 
+  RECEIVE_FINAL_NOTIFY            : 5,
+  RECEIVE_BAD_NOTIFY              : 6 
+};
+
+/**
+ * It's implementation of RFC 6665 Subscriber
+ */
+module.exports = class Subscriber extends EventEmitter 
+{
+  /**
+   * Expose C object.
+   */
+  static get C()
+  {
+    return C;
+  }
+
+  get C()
+  {
+    return C;
+  }
+
+  /**
+   * -param {Object} ua reference to JsSIP.UA
+   * -param {String} target
+   * -param {Object} options 
+   *   -param {String} eventName Event header value. May end with optional ;id=xxx
+   *   -param {String} accept Accept header value
+   *   -param {Number} expires Expires header value. Optional. Default is 900
+   *   -param {String} contentType Content-Type header value
+   *   -param {String} allowEvents Allow-Events header value. Optional.
+   *   -param {Object} params Will have priority over ua.configuration. Optional.
+   *      If set please define: to_uri, to_display_name, from_uri, from_display_name
+   *   -param {Array}  headers Optional. Additional SIP headers.
+   *   -param {Object} credential. Will have priority over ua.configuration. Optional.
+   */
+  constructor(ua, target, { eventName, accept, expires, contentType, 
```

Comment:
As in `Notifier`, let's only put the optional values in the options paramter.

Reply:
OK

Reply:
OK, the same for Subscriber constructor.